src/EditorFeatures/CSharpTest/CodeActions/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -2345,5 +2345,82 @@ void M()
 }
 ");
         }
+
+        [WorkItem(1130990)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void InParentConditionalAccessExpressions()
+        {
+            var code =
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var y = [|F(new C())|]?.F(new C())?.F(new C());
+        return x;
+    }
+}";
+
+            var expected =
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var {|Rename:c|} = F(new C());
+        var y = c?.F(new C())?.F(new C());
+        return x;
+    }
+}";
+
+            Test(code, expected, index: 0, compareTokens: false);
+        }
+
+        [WorkItem(1130990)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void MissingAcrossMultipleParentConditionalAccessExpressions()
+        {
+            TestMissing(
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var y = [|F(new C())?.F(new C())|]?.F(new C());
```

Comment:
That is because how the tree is structured in ConditionalAccessExpression cases.

[|F(New C)?.F(New C)|]?.F(New C) is an invalid selection. The selection cannot be represented as a single expression in the syntax tree. They are two invocation expressions under two different parent conditional access expressions. Introduce variable does not expand selection, so, in this case, the selection is just invalid. 

Here is a comparison screenshot of syntax trees for `F().F().F()` and `F()?.F()?.F()`

![image](https://cloud.githubusercontent.com/assets/4213867/7424233/c3efde12-ef53-11e4-9fa7-416fd670f69e.png)

That is my understanding. :smiley: